workflows/diagnostics/fv3net/diagnostics/offline/_input_sensitivity.py

```diff
@@ -0,0 +1,215 @@
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import numpy as np
+import logging
+from typing import Mapping, Tuple, Sequence, Iterable
+import xarray as xr
+
+import fv3fit
+from fv3fit.sklearn._random_forest import SklearnWrapper
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+JACOBIAN_NAME = "jacobians.png"
+RF_FEATURE_IMPORTANCE_NAME = "rf_feature_importances.png"
+
+
+def dataset_to_dict_input(ds):
+    if "time" in ds.dims:
+        ds = ds.isel(time=0).squeeze(drop=True)
+    stacked = ds.stack(sample=["tile", "x", "y"]).transpose("sample", ...)
+    data = {}
+    for var in stacked:
+        values = stacked[var].values
+        if len(stacked[var].dims) == 1:
+            values = values.reshape(-1, 1)
+        data[var] = values
+    return data
+
+
+def _count_features_2d(
+    quantity_names: Iterable[str], dataset: xr.Dataset, sample_dim_name: str
+) -> Mapping[str, int]:
+    """
+    count features for (sample[, z]) arrays.
+    Copied from fv3fit._shared.packer, as this logic is pretty robust.
+    """
+    for name in quantity_names:
+        if len(dataset[name].dims) > 2:
+            value = dataset[name]
+            raise ValueError(
+                "can only count 1D/2D (sample[, z]) "
+                f"variables, recieved values for {name} with dimensions {value.dims}"
+            )
+    return_dict = {}
+    for name in quantity_names:
+        value = dataset[name]
+        if len(value.dims) == 1 and value.dims[0] == sample_dim_name:
+            return_dict[name] = 1
+        elif value.dims[0] != sample_dim_name:
+            raise ValueError(
+                f"cannot count values for {name} whose first dimension is not the "
+                f"sample dimension ({sample_dim_name}), has dims {value.dims}"
+            )
+        else:
+            return_dict[name] = value.shape[1]
+    return return_dict
+
+
+def _get_variable_indices(
+    data: xr.Dataset, variables: Sequence[str]
+) -> Mapping[str, Tuple[int, int]]:
+    if "time" in data.dims:
+        data = data.isel(time=0).squeeze(drop=True)
+    stacked = data.stack(sample=["tile", "x", "y"])
+    variable_dims = _count_features_2d(
+        variables, stacked.transpose("sample", ...), "sample"
+    )
+    start = 0
+    variable_indices = {}
+    for var, var_dim in variable_dims.items():
+        variable_indices[var] = (start, start + var_dim)
+        start += var_dim
+    return variable_indices
+
+
+def plot_input_sensitivity(
```

Comment:
I realize this function is not new, but it does induce tight coupling between `fv3fit` and these offline diags. Would be great to use polymorphism for this behavior rather than the brittle exception catching logic below. Can you add some interfaces to `fv3fit.Predictor` that help with this? e.g. `fv3fit.Predictor.get_input_sensitivity`?

Reply:
As discussed I'll punt on this comment for this PR but am tracking it in an issue https://github.com/ai2cm/fv3net/issues/1596 since the changes would probably be a large diff.